### ISSUE TITLE
Migrate from nbsphinx to myst, myst-nb

### ISSUE THREAD

**Issue by dcherian** [pydata/xarray#7924]
### Is your feature request related to a problem?

I think we should switch to MyST markdown for our docs. I've been using MyST markdown and MySTNB in docs in other projects and it works quite well. 

Advantages:
1. We get HTML reprs in the docs ([example](https://cf-xarray.readthedocs.io/en/latest/selecting.html)) which is a big improvement. (#6620)
2. I think many find markdown  a lot easier to write than RST

There's a tool to migrate RST to MyST ([RTD's migration guide](https://docs.readthedocs.io/en/stable/guides/migrate-rest-myst.html)).

**Comment by shoyer** [comment]
I agree! I [use MySt and MyST-NB](https://github.com/google/xarray-beam/tree/main/docs) for Xarray-Beam and it works quite well.

**Comment by DocOtak** [comment]
Would docstrings in the code also be written in MyST?

**Comment by shoyer** [comment]
> Would docstrings in the code also be written in MyST?

This would be really nice, but I don't think there is good tooling for it yet. We use [napoleon](https://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html) for parsing docstrings in our docs, which supports the NumPy and Google docstring conventions. We currently use the NumPy docstring convention which expects RST.

The Google docstring convention is rather different (I actually like it better), and I've seen people use markdown with it before. But I don't think tools like Napolean support that (yet).

**Comment by benbovy** [comment]
I also use MyST and MyST-NB and I like it a lot!

It seems that the maintainer MyST-NB has stepped down from that project as well as a few other projects (including the tool to migrate from RST to MyST), though, which causes some issues of dependencies that are still upper bounded to old versions (e.g., sphinx < 6, myst-parser, ipython, etc. See all [open PRs](https://github.com/executablebooks/MyST-NB/pulls)). It is not yet clear to me if someone else will take over the maintenance in the short term.

**Comment by VeckoTheGecko** [comment]
> It seems that the maintainer MyST-NB has stepped down from that project

Looks like the maintainership status of MyST-NB is now in good standing

**Comment by VeckoTheGecko** [comment]
Does anyone have a suggestion on how to proceed with this in a reviewable way?

I was thinking:
- PR1: Add support for myst-parser (https://github.com/pydata/xarray/pull/11167 )
- PR2: Migrate from nbsphinx to myst-nb (converting the doc pages which use nbsphinx along the way)
- PRx: Gradually migrate the rest of the doc pages to mystnb

How does that sound? I'm concerned about merge conflicts negatively affecting [PRs labeled topic-documentation](https://github.com/pydata/xarray/pulls?q=sort%3Aupdated-desc+is%3Apr+is%3Aopen+label%3Atopic-documentation+)

Additional questions:
- Which files do we not want to convert? (e.g., `doc/_templates`? `doc/whats-new`?)

**Comment by shoyer** [comment]
I think we should try to convert every `.rst` doc page to `.md` and every `.ipynb` to to Myst-NB. `doc/_templates` is something special for Sphinx auto doc summaries, so we can skip that.

If automatic converters don't work well, I think AI tools like Claude Code could probably power through most of this pretty efficiently. I would definitely start doing this incrementally if you're using manual processes, but with AI assistants iterating on larger changes is also possible -- just include your "plan" in the PR description and we can iterate on that.

**Comment by VeckoTheGecko** [comment]
> and every `.ipynb` to to Myst-NB

I assume you mean doc pages that use `nbsphinx` (`jupyter-execute` blocks)? The only notebooks I can see are the gallery examples which I don't think would benefit much from being converted?

**Comment by shoyer** [comment]
> > and every `.ipynb` to to Myst-NB
> 
> I assume you mean doc pages that use `nbsphinx` (`jupyter-execute` blocks)? The only notebooks I can see are the gallery examples which I don't think would benefit much from being converted?

Yes, that was want I was thinking. Though indeed maybe this is not necessary after all.

**Comment by VeckoTheGecko** [comment]
OK - https://github.com/pydata/xarray/pull/11456 focussed on the pages using the old nbsphinx plugin. Now that's merged we are most of the way there. Remaining items are

- [ ] Conversion of the remaining .rst files - migrate these where possible
- [ ] We still have a bunch of `eval-rst` blocks in the markdown which can be substituted out for myst syntax
- [ ] There are still some lingering `.. jupyter-execute::` blocks in `terminology.rst`. I'm not sure how this terminology section works and how it would look like in Myst - I'm not sure if its possible to get rid of these
